ManalHamdi/assignment3/Assignment3.cpp

```diff
@@ -0,0 +1,88 @@
+//Assumption1: the function should be case insensitive because in real world
+#include <iostream>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+class Dictionary {
+	private:
+		std::set<std::string> words;
+	public:
+		Dictionary(const std::vector<std::string> words) {
+			for (auto word : words) {
+				this->words.insert(word);
+			}
+		}
+		bool isWord(const std::string word) {
+			return words.count(word);
+		}
+		bool isPrefix(const std::string big, const std::string small) {
+			if (big.size() >= small.size()) {
+				return big.substr(0, small.size()) == small;
+			}
+			return false;
+		}
+		bool isPrefix(const std::string prefix) {
+			for (auto word : words) {
+				if (isPrefix(word, prefix)) {
+					return true;
+				}
+			}
+			return false;
+		}
+};
+class Grid {
+public:
+	std::vector<std::vector<char>> grid;
```

Comment:
You should never expose internal structure of a class. Please move `grid`, `number_rows` and `number_columns` to `private` and provide a constructor to initialise them, similarly to how you've structured `Dictionary`.